Library/Source/NativeEngine.cpp

```diff
@@ -277,6 +282,109 @@ namespace babylon
 
+        class ViewClearState
+        {
+        public:
+            ViewClearState(uint16_t viewId)
+                : m_viewId{ viewId }
+            {}
+
+            bool Update(const Napi::CallbackInfo& info)
+            {
+                auto r = info[0].As<Napi::Number>().FloatValue();
+                auto g = info[1].As<Napi::Number>().FloatValue();
+                auto b = info[2].As<Napi::Number>().FloatValue();
+                auto a = info[3].IsUndefined() ? 1.f : info[3].As<Napi::Number>().FloatValue();
+                auto backBuffer = info[4].IsUndefined() ? true : info[4].As<Napi::Boolean>().Value();
+                auto depth = info[5].IsUndefined() ? true : info[5].As<Napi::Boolean>().Value();
+                auto stencil = info[6].IsUndefined() ? true : info[6].As<Napi::Boolean>().Value();
+
+                bool needToUpdate = r != m_red
+                    || g != m_green
+                    || b != m_blue
+                    || a != m_alpha
+                    || backBuffer != m_backBuffer
+                    || depth != m_depth
+                    || stencil != m_stencil;
+                if (needToUpdate)
+                {
+                    m_red = r;
+                    m_green = g;
+                    m_blue = b;
+                    m_alpha = a;
+                    m_backBuffer = backBuffer;
+                    m_depth = depth;
+                    m_stencil = stencil;
+
+                    Update();
+                }
+
+                return needToUpdate;
+            }
+
+            void Update() const
+            {
+                // TODO: Backbuffer, depth, and stencil.
+                bgfx::setViewClear(m_viewId, BGFX_CLEAR_COLOR | (m_depth ? BGFX_CLEAR_DEPTH : 0x0), Color());
+            }
+
+        private:
+            const uint16_t m_viewId{};
+            float m_red{ 68.f / 255.f };
+            float m_green{ 51.f / 255.f };
+            float m_blue{ 85.f / 255.f };
+            float m_alpha{ 1.f };
+            bool m_backBuffer{ true };
+            bool m_depth{ true };
+            bool m_stencil{ true };
+
+            uint32_t Color() const
+            {
+                uint32_t color = 0x0;
+                color += static_cast<uint8_t>(m_red * std::numeric_limits<uint8_t>::max());
+                color = color << 8;
+                color += static_cast<uint8_t>(m_green * std::numeric_limits<uint8_t>::max());
+                color = color << 8;
+                color += static_cast<uint8_t>(m_blue * std::numeric_limits<uint8_t>::max());
+                color = color << 8;
+                color += static_cast<uint8_t>(m_alpha * std::numeric_limits<uint8_t>::max());
+                return color;
+            }
+        };
+
+        struct FrameBufferData final
+        {
+            FrameBufferData(bgfx::FrameBufferHandle frameBuffer, uint16_t width, uint16_t height)
+                : FrameBuffer{ frameBuffer }
+                , ViewId{ NextViewId++ }
```

Comment:
Check the limit for the viewid (max 256 if I recall correctly the default value). But you can use the same viewid for all the render target rendering if you call bgfx::frame when unbinding the target. The drawcalls will be issued at that time and you are free to use the same viewid.

Reply:
And no swap chain if view 0 is untouched. At the same time, maybe you can use a table of allocated viewids like bgfx does with allocated framebuffer.

Reply:
Good call, there's a max value in the caps I can query. I've added an assert to do that, will come up with the next push. As for calling frame() every time, do we want to do that? I'm not 100% clear on all the moving pieces involved, but it seems like that solution doesn't quite jive with BGFX's existing utilities for views and draw call batching, so I'm wondering if there's a reason not to.

Reply:
256 viewed for all the application life-time seems dangerous. What to do when you reach that value? I encounter that same issue with my pet project and that was the answer Branimir gave me https://github.com/bkaradzic/bgfx/issues/1866#issuecomment-523726872

Reply:
Spoke offline, added the recycling mechanism we discussed there.